packages/compass-crud/src/components/change-view/shape-utils.ts

```diff
@@ -0,0 +1,16 @@
+export function isSimpleObject(value: any) {
+  return (
+    Object.prototype.toString.call(value) === '[object Object]' &&
+    !value._bsontype
+  );
+}
+
+export function getValueShape(value: any) {
```

Comment:
```suggestion
export function getValueShape(value: any): 'array' | 'object' | 'leaf' {
```

For autocompletion and compiler checks.